curated_transformers/tests/models/util.py

```diff
@@ -9,20 +9,152 @@
 from ..util import torch_assertclose
 
 
+def assert_causal_lm_output_equals_hf(
+    model_class: Type[FromHFHub],
+    model_name: str,
+    torch_device: torch.device,
+    *,
+    atol=1e-5,
+    rtol=1e-5,
+    model_revision: str = "main",
+    with_torch_compile: bool = False,
+):
+    model = model_class.from_hf_hub(
+        name=model_name, revision=model_revision, device=torch_device
+    )
+    model.eval()
+
+    for _, param in model.state_dict().items():
+        assert param.device == torch_device
+
+    if with_torch_compile:
+        model = torch.compile(model)
+
+    hf_model = transformers.AutoModelForCausalLM.from_pretrained(
+        model_name,
+        revision=model_revision,
+    )
+    hf_model.eval()
+    hf_model.to(torch_device)
+
+    torch.manual_seed(0)
+    X = torch.randint(0, hf_model.config.vocab_size, (2, 10), device=torch_device)
+
+    with torch.no_grad():
+        Y = model(X).logits
+        Y_hf = hf_model(X).logits
+    torch_assertclose(Y, Y_hf, atol=atol, rtol=rtol)
+
+    mask = torch.rand((2, 10), dtype=torch.float, device=torch_device) < 0.5
```

Comment:
Do we need to reset the manual seed for this invocation as well? I recall that we do that in the sampling generator tests.

Reply:
I considered doing this, since it's a known failure. However, the problem is that if Torch fixes this issue, the test will suddenly pass (and thus fail), which is probably not nice.

Reply:
That is so that we can reorder/remove particular generation tests in the future without failing the generation tests after that. But that's not an issue here, since we do not manually validate the output.